Sources/LiveViewNative/Coordinators/LiveViewCoordinator.swift

```diff
@@ -0,0 +1,332 @@
+    public func pushEvent(type: String, event: String, value: Any) async throws {
+        // isValidJSONObject only accepts objects, but we want to check that the value can be serialized as a field of an object
+        precondition(JSONSerialization.isValidJSONObject(["a": value]))
+        try await doPushEvent("event", payload: [
+            "type": type,
+            "event": event,
+            "value": value,
+        ])
+    }
+    
+    internal func doPushEvent(_ event: String, payload: Payload) async throws {
+        guard let channel = channel else {
+            return
+        }
+        
+        let token = self.currentConnectionToken
+
+        let replyPayload = try await withCheckedThrowingContinuation({ continuation in
+            channel.push(event, payload: payload, timeout: PUSH_TIMEOUT)
+                .receive("ok") { reply in
+                    continuation.resume(returning: reply.payload)
+                }
+                .receive("error") { message in
+                    continuation.resume(throwing: LiveConnectionError.eventError(message))
+                }
+        })
+        
+        guard currentConnectionToken === token else {
+            // TODO: maybe this should just silently fail?
+            throw CancellationError()
+        }
+        
+        if let diffPayload = replyPayload["diff"] as? Payload {
+            do {
+                try self.handleDiff(payload: diffPayload, baseURL: self.url)
+            } catch {
+                fatalError("todo")
+            }
+        } else if session.config.liveRedirectsEnabled,
+                  let redirect = (replyPayload["live_redirect"] as? Payload).flatMap({ LiveRedirect(from: $0, relativeTo: self.url) }) {
+            await session.redirect(redirect)
+        }
+    }
+    
+    private func handleDiff(payload: Payload, baseURL: URL) throws {
+        if session.config.eventHandlersEnabled,
+           let events = payload["e"] as? [[Any]] {
+            for event in events {
+                guard let name = event[0] as? String,
+                      let value = event[1] as? Payload,
+                      let handler = eventHandlers[name] else {
+                    continue
+                }
+                handler(value)
+            }
+        }
+        let diff = try RootDiff(from: FragmentDecoder(data: payload))
+        self.rendered = try self.rendered.merge(with: diff)
+        self.document?.merge(with: try Document.parse(self.rendered.buildString()))
+    }
+    
+    private nonisolated func parseDOM(html: String, baseURL: URL) throws -> Elements {
+        // we parse the DOM as XML rather than HTML, because we don't want it to apply browser-specific HTML transformations
+        // such as <image> -> <img>
+        let document = try Parser.xmlParser().parseInput(html, baseURL.absoluteString)
```

Comment:
This can actually be changed back to the HTML parser, since that's all it's used for, now that core handles parsing the live view.